charts/datadog/templates/_otel_agent_config.yaml

```diff
@@ -1,5 +1,43 @@
 {{- define "otel-agent-config-configmap-content" -}}
 otel-config.yaml: {{- if .Values.datadog.otelCollector.config }} {{ toYaml .Values.datadog.otelCollector.config | indent 4 }}
+  {{- else if .Values.otelAgentGateway.enabled }} |
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+             endpoint: 0.0.0.0:{{ include "get-port-number-from-name" (dict "ports" .Values.datadog.otelCollector.ports "portName" "otel-grpc") }}
+          http:
+             endpoint: 0.0.0.0:{{ include "get-port-number-from-name" (dict "ports" .Values.datadog.otelCollector.ports "portName" "otel-http") }}
+    exporters:
+      debug:
+        verbosity: detailed
+      otlphttp:
+        endpoint: http://{{ template "datadog.fullname" . }}-otel-agent-gateway:4318
```

Comment:
Not sure to understand what we are doing here. If both otel agent and otel gateway agent is configured, then we will automatically point the "regular" otel agents to the gateway otel agent ?

Reply:
That's right, this config represents the arrow from the blue otel-agent to the LB: https://docs.google.com/presentation/d/1DjXetxS0HEAcozuaX2XTXvr_0kbyFkSd29wxAsY_IgY/edit?slide=id.g37cb75592a0_0_15#slide=id.g37cb75592a0_0_15